packages/actor-init-sparql/test/system/Engine-test.ts

```diff
@@ -86,6 +89,68 @@ describe('System test: ActorInitSparql', () => {
         expect((await arrayifyStream((<IQueryResultBindings> await engine.query(query, context)).bindingsStream)))
           .toEqual([]);
       });
+
+      describe('custom function', () => {
+        const funcAllow = 'allowAll';
+        const baseFunctionName = (funcName: string) =>
+          `http://example.org/functions#${funcName}`;
+        const baseQuery = (funcName: string) => `PREFIX func: <http://example.org/functions#>
+        SELECT * WHERE {
+              ?s ?p ?o.
+            FILTER (func:${funcName}(?o))
+        }`;
+        const baseFunctionMap = new Map<string, (args: RDF.Term[]) => Promise<RDF.Term>>([
+          [ baseFunctionName(funcAllow), async(arg: RDF.Term[]) => {
+            return literal('true', 'http://www.w3.org/2001/XMLSchema#boolean');
+          } ],
+        ]);
+        const baseFunctionCreator = (functionName: RDF.NamedNode) => {
+          return (arg: RDF.Term[]) => {
+            return literal('true', 'http://www.w3.org/2001/XMLSchema#boolean');
+          };
+        };
+
+        it('rejects when map does not match', async() => {
+          const context = <any> { sources: [ 'https://www.rubensworks.net/' ]};
```

Comment:
I would suggest running these tests on a custom N3Store, with only a very small set of triples, so we can really do in-depth checks to see which triples come out of the filter, instead of just counting the number of results.